### ISSUE TITLE
Git storage?

### ISSUE THREAD

**Issue by bbinet** [scragg0x/realms-wiki#47]
Sorry for creating an issue to ask questions, but I couldn't find a mailing list for the realms-wiki project.

Thanks for realms-wiki which sounds great and has a very appealing web interface!

I was wondering why realms-wiki needs a database (sqlite, mysql or postgres) if it is a Git based wiki: does realms-wiki really use a Git repository as storage?

I'm also interested in updating realms-wiki pages directly from Git cli (without using the web interface): does realms-wiki supports updating the wiki when we git push our changes to a repository (like Gollum)?

Last question, on http://realms.io/ the title is "Realms Wiki Beta": do you consider it as beta software, or can it be run in production? Is the project still active, and is there a roadmap somewhere?

Thanks!

**Comment by CristianCantoro** [comment]
Hi,

I am new to the project myself, and I had the same questions. So I have searched a little, here's my answers:

* As it is stated in the [project's requirements](http://realms.io#requirements): `Anon or single user does not require a database.`. So the database is used only to manage authentication.

* From the [config file](https://github.com/scragg0x/realms-wiki/blob/master/realms/config/__init__.py#L118) you can see that the default location for the git repo is in `/tmp/wiki/`.
I have tried to manually make changes to the files and commit them and it woks.
So I think that yes, wth a little configuration you can set up your realms-wiki installation and accesses to edit directly the repo.

I can not help you with the third question, though.

**Comment by loleg** [comment]
In addition to what @CristianCantoro completely correctly said, the only problem I have encountered so far with manually adding/updating/deleting files in the Git repo, is that timestamps are reset - leading to an INDEX page where all pages have the same date - an issue I raised in #45.

Only time, testing and the experience of users like ourselves can adequately determine how stable the project really is. I've had Realms in semi-production now for 3 months and am happy to report it has been a useful and reliable tool. 

Except for one hiccup: of my users lost work repeatedly by forgetting to click *Publish* - which I eventually diagnosed as #44, but too late to avoid eroding that particular user's faith. This is representative of the fact that the wiki could use additional safeguards to ensure user data is protected at all cost, and what I would base a decision on to go into production. 

Plus, as far as I know, scalability with a lot of simultaneous users has not been tested. You will definitely have issues at this stage if people do not coordinate efforts and edit the same page without using collaboration mode. To get around this, I would suggest having discrete wikis for small sub-groups, whose work is merged from time to time using pull requests.
